### PR TITLE
Remove macros

### LINKED ISSUE #47: Remove ARGCHK1/2/3
It would be nice if we could remove the following code in `j.h`:
```cpp
#define DEADARG(x)      0
#define ARGCHK1D(x)
#define ARGCHK2D(x,y)

#define ARGCHK1(x)      RZ(x) DEADARG(x);   // bit set in deadbeef
#define ARGCHK2(x,y)    ARGCHK1(x) ARGCHK1(y)
#define ARGCHK3(x,y,z)  ARGCHK1(x) ARGCHK1(y) ARGCHK1(z)
```
and all of the places that call it (~800). 

Not sure it is possible though as some unit tests might be testing failures (so might have to replace with something).

### PR DESCRIPTION
**Pull request by juntuu**
Removed unused macros.

There are still some macros that are only used via token pasting, and thus appear unused when searching directly:
- \<type\>##SIZE
- \<name\>##values
- VA[12]##\<id\>

Also removed `ARGCHK*D` macros and expanded & removed `ARGCHK*` macros; resolves #47, also part of #13.

### CODE REVIEW

File: jsrc/adverbs/a.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ static A swap2(J jt,A a,A w,A self){DECLF; F2PREFIP; jtinplace = (J)(intptr_t)((
 
 // w~, which is either reflexive/passive or evoke
  A jtswap(J jt, A w){A y;C*s;I n;
- ARGCHK1(w); 
+ if(!w) return 0;
```
Comment (codereport) [inline jsrc/adverbs/a.c:20]: Ah, sorry. The original issue is unclear. For the `ARGCHK` macros, we actually want to just delete them, not replace them with anything. Is that possible?

**Review by codereport — CHANGES_REQUESTED** [review]

File: jsrc/adverbs/a.c (hunk 1 above)
Comment (juntuu) [reply jsrc/adverbs/a.c:20]: That was what I gathered from the issue, however it seems at least some functionality relies on the check.

```
98% tests passed, 7 tests failed out of 396

Total Test time (real) = 296.12 sec

The following tests FAILED:
        132 - g232 (SEGFAULT)
        179 - g420stch (SEGFAULT)  -- previously failing on osx
        237 - g631 (SEGFAULT)
        268 - gf (SEGFAULT)
        315 - gq (SEGFAULT)
        387 - gu (SEGFAULT)
        388 - gu0 (SEGFAULT)
Errors while running CTest
```

I'll check, if I can remove some and only leave those that are needed for the existing tests.

The commented out tests also add one more unknown to this.

Comment (codereport) [reply jsrc/adverbs/a.c:20]: Yea, I would say let's remove as much as possible.

**Comment by juntuu** [comment]
I got quite a few uses removed, but some places needed the check.

**Comment by Sebanisu** [comment]
I think some of the maros on my list are removed in this pull request https://github.com/codereport/jsource/issues/60
If you got all 10 or the remaining ones, we can close my issue.

**Review by codereport — APPROVED** [review]
:fire: :rocket:

**Pull request merged**